tests/unittest/St30pHandler.cpp

```diff
@@ -0,0 +1,254 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright(c) 2025 Intel Corporation
+ */
+
+#include "noctx.hpp"
+
+St30pHandler::St30pHandler(st_tests_context* ctx, SessionUserData* sessionUserData,
+                           st30p_tx_ops ops_tx, st30p_rx_ops ops_rx,
+                           uint msPerFramebuffer, bool create, bool start)
+    : Handlers(ctx, sessionUserData, AUDIO_CLOCK_HRTZ),
+      msPerFramebuffer(msPerFramebuffer) {
+  if (ops_tx.name == nullptr || ops_rx.name == nullptr) {
+    fillSt30pOps();
+    ops_tx = sessionsOpsTx;
+    ops_rx = sessionsOpsRx;
+  } else {
+    sessionsOpsTx = ops_tx;
+    sessionsOpsRx = ops_rx;
+  }
+
+  EXPECT_TRUE(sessionUserData != nullptr);
+  if (!sessionUserData) return;
```

Comment:
I think it should `throw` an exception here. this seems to be error in test definition not a failed test.